### PR TITLE
[DevOps] Snyk Integration

### PR DESCRIPTION
**Pull request by ChathurindaRanasinghe**
**Description**

- Fixed deprecation issues with the `set-output` command. - https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
- Changed the organisation to [WAVE](https://app.snyk.io/org/wave)
- Updated GitHub Action versions for Python.
- Fixed the issue with Go dependency scanning failure.
- Updated the [changed-files](https://github.com/marketplace/actions/changed-files) GitHub Action version.
- `remote-repo-url` will follow `repo/branch` format & `project-name` will follow `PRODUCT/repo/branch/file` format.
- Added a .snyk file to ignore false license issues.

**Parent Issues**

- https://github.com/h2oai/h2o-ops/issues/285
- https://github.com/h2oai/h2o-ops/issues/241

### CODE REVIEW

**Review by lo5 — DISMISSED** [review]

**Review by mturoci — CHANGES_REQUESTED** [review]
Thanks @ChathurindaRanasinghe. Unfortunately, it seems like this PR doesn't solve the problem. I created a new branch off this one and made a new PR to verify this works, but keep getting the same error - https://github.com/h2oai/wave/actions/runs/4352959484/jobs/7606400169.

File: .github/workflows/snyk-scan.yml [hunk 1]
```diff
@@ -92,40 +96,43 @@ jobs:
...
         id: extract_ref
 
       - uses: actions/checkout@master
 
-      - name: Snyk scan for Go dependancies
-        uses: snyk/actions/golang@master
+      - uses: snyk/actions/setup@master
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "1.15"
```
[the remark this answers is not in the document]
Comment (ChathurindaRanasinghe) [reply .github/workflows/snyk-scan.yml:107]: Sure, I will update the version.

**Comment by ChathurindaRanasinghe** [comment]
> Thanks @ChathurindaRanasinghe. Unfortunately, it seems like this PR doesn't solve the problem. I created a new branch off this one and made a new PR to verify this works, but keep getting the same error - https://github.com/h2oai/wave/actions/runs/4352959484/jobs/7606400169.

I will add a Snyk policy file to ignore those errors.

File: .github/workflows/snyk-scan.yml [hunk 2]
```diff
@@ -29,16 +29,20 @@ jobs:
             py/examples/requirements.txt
             py/Makefile
 
-      - name: Snyk scan for Go dependancies
+      - uses: snyk/actions/setup@master
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "1.15"
```
[the remark this answers is not in the document]
Comment (ChathurindaRanasinghe) [reply .github/workflows/snyk-scan.yml]: done

**Comment by ChathurindaRanasinghe** [comment]
image

cc: @mturoci

**Review by mturoci — APPROVED** [review]

**Pull request merged**